ToNote/Controls/NoteRichTextBoxPanel.cs

```diff
@@ -0,0 +1,186 @@
+
+                foreach (var item in e.NewItems)
+                {
+                    if (item is ExtendedRichTextBox rtb)
+                        ConfigureRichTextBoxEvents(rtb);
+                }
+            };
+        }
+
+        public Note Note
+        {
+            get { return (Note)GetValue(NoteProperty); }
+            set { SetValue(NoteProperty, value); }
+        }
+
+        // On a new value bound to Note, generates textboxes for each file the note has and fills them with their respective content.
+        public static readonly DependencyProperty NoteProperty = DependencyProperty.Register("Note",
+           typeof(Note), typeof(NoteRichTextBoxPanel), new FrameworkPropertyMetadata(null) { PropertyChangedCallback = (s, e) =>
+           {
+               if (!(e.NewValue is Note NewNoteValue) || NewNoteValue == null) return;
+
+               var panel = (NoteRichTextBoxPanel)s;
+
+               panel.Items.Clear();
+
+               if (NewNoteValue != null)
+               {
+                   foreach (var file in (NewNoteValue.FileNames))
+                   {
+                       var rtb = new ExtendedRichTextBox()
+                       { Style = App.Current.TryFindResource("NoteContentRichTextBoxStyle") as Style };
+
+                       rtb.ReadFromFile(file);
+
+                       panel.Items.Add(rtb);
+                   }
+               }
+           }
+           });
+
+        //Command to add a new TextBox. Implemented so the command can be invoked from XAML instead of back-end;
+        public ICommand AddRichTextBoxCommand
+        {
+            get { return (ICommand)GetValue(AddRichTextBoxCommandProperty); }
+            set { SetValue(AddRichTextBoxCommandProperty, value); }
+        }
+
+        public static readonly DependencyProperty AddRichTextBoxCommandProperty = DependencyProperty.Register("AddRichTextBoxCommand",
+           typeof(ICommand), typeof(NoteRichTextBoxPanel), new FrameworkPropertyMetadata(new RelayCommand<NoteRichTextBoxPanel>((panel) => 
+           {
+               panel.Items.Add(new ExtendedRichTextBox() { Style = App.Current.TryFindResource("NoteContentRichTextBoxStyle") as Style });
+           })));
+
+        //Command to save each ExtendedRichTextBox control's content's to a respective .rtf file
+        public ICommand SaveContentsToFilesCommand
+        {
+            get { return (ICommand)GetValue(SaveContentsToFilesCommandProperty); }
+            set { SetValue(SaveContentsToFilesCommandProperty, value); }
+        }
+
+        public static readonly DependencyProperty SaveContentsToFilesCommandProperty = DependencyProperty.Register("SaveContentsToFilesCommand",
+           typeof(ICommand), typeof(NoteRichTextBoxPanel), new FrameworkPropertyMetadata(new RelayCommand<NoteRichTextBoxPanel>((panel) =>
+           {
+               var note = panel?.Note;
+
+               if (panel == null || note == null) return;
```

Comment:
`panel == null` check is redundat, if panel is null, note will be null as well, so only checking note is enough